include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.td

```diff
@@ -102,6 +102,18 @@ def TTS_MakeTensorPtrOp
       Builder b(getContext());
       return ::mlir::getMixedValues(getStaticShape(), getShape(), b);
     }
+    bool isBlockPtr() {
+      return !getOrder().empty();
+    }
+    bool isRegularPtr() {
+      return !isBlockPtr() &&
+             llvm::all_of(getStaticShape(), [](auto shape) { return shape == 0; });
+    }
+    bool isSplitPtr() {
+      return !isBlockPtr() &&
+             !isRegularPtr() &&
+             llvm::any_of(getStaticShape(), [](auto shape) { return shape != 0; });
```

Comment:
1. the term a "regular pointer" is a bit vague, is this funciton necessary?

2. is "not-regular" the same as "any-shape-not-0"? Looks redundant to me at first glance:


!isRegular ==> isBlockPtr || not-all-shape-0
we already weeded out is blockPtr so... 
isBlockPtr || not-all-shape-0 ==> not-all-shape-0 ==> any-shape-non-0

so i _think_ `isSplitPtr()` is just `!isBlockPtr && !isRegularPtr`

Reply:
I just renamed "regular" to "structured" and simplified the conditions!